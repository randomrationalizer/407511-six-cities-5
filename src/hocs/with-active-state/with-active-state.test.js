import React from "react";
import PropTypes from "prop-types";
import renderer from "react-test-renderer";
import withActiveState from "./with-active-state";


const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withActiveState(MockComponent);

it(`Should withActiveState component renders correctly`, () => {
  const tree = renderer
    .create(
        <MockComponentWrapped
          isActive={true}
        >
          <React.Fragment />
        </MockComponentWrapped>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
