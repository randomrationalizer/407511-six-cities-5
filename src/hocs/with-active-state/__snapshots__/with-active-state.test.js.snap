// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should withActiveState component renders correctly 1`] = `<div />`;
